tsp-output/@azure-tools/typespec-csharp/src/Generated/Internal/ClientUriBuilder.cs

```diff
@@ -1,210 +1,82 @@
-        }
-
-        public void AppendQuery(string name, string value, bool escape)
-        {
-            Argument.AssertNotNullOrWhiteSpace(name, nameof(name));
-            Argument.AssertNotNullOrWhiteSpace(value, nameof(value));
-
-            if (QueryBuilder.Length == 0)
-            {
-                QueryBuilder.Append('?');
-            }
-            else
-            {
-                QueryBuilder.Append('&');
-            }
-
-            if (escape)
-            {
-                value = Uri.EscapeDataString(value);
-            }
-
-            QueryBuilder.Append(name);
-            QueryBuilder.Append('=');
-            QueryBuilder.Append(value);
-        }
+internal class ClientUriBuilder
+{
+    private UriBuilder? _uriBuilder;
+    private StringBuilder? _pathBuilder;
+    private StringBuilder? _queryBuilder;
 
-        public void AppendQuery(string name, bool value, bool escape = false)
-        {
-            AppendQuery(name, ModelSerializationExtensions.TypeFormatters.ConvertToString(value), escape);
-        }
+    private UriBuilder UriBuilder => _uriBuilder ??= new();
+    private StringBuilder PathBuilder => _pathBuilder ??= new(UriBuilder.Path);
+    private StringBuilder QueryBuilder => _queryBuilder ??= new(UriBuilder.Query);
 
-        public void AppendQuery(string name, float value, bool escape = true)
-        {
-            AppendQuery(name, ModelSerializationExtensions.TypeFormatters.ConvertToString(value), escape);
-        }
+    public ClientUriBuilder()
+    {
+    }
 
-        public void AppendQuery(string name, DateTimeOffset value, string format, bool escape = true)
-        {
-            AppendQuery(name, ModelSerializationExtensions.TypeFormatters.ConvertToString(value, format), escape);
-        }
+    public void Reset(Uri uri)
+    {
+        _uriBuilder = new(uri);
+        _pathBuilder = new(_uriBuilder.Path);
+        _queryBuilder = new(_uriBuilder.Query);
+    }
 
-        public void AppendQuery(string name, TimeSpan value, string format, bool escape = true)
-        {
-            AppendQuery(name, ModelSerializationExtensions.TypeFormatters.ConvertToString(value, format), escape);
-        }
+    public void AppendPath(string value, bool escape)
+    {
+        Argument.AssertNotNullOrWhiteSpace(value, nameof(value));
 
-        public void AppendQuery(string name, double value, bool escape = true)
+        if (escape)
         {
-            AppendQuery(name, ModelSerializationExtensions.TypeFormatters.ConvertToString(value), escape);
+            value = Uri.EscapeDataString(value);
         }
 
-        public void AppendQuery(string name, decimal value, bool escape = true)
-        {
-            AppendQuery(name, ModelSerializationExtensions.TypeFormatters.ConvertToString(value), escape);
-        }
+        PathBuilder.Append(value);
+        UriBuilder.Path = PathBuilder.ToString();
```

Comment:
comparing with the old code, we removed the `pathSeparator` part in this method. This is included in this PR: https://github.com/Azure/autorest.csharp/pull/4519

Reply:
I'm not sure I fully understand your question or the concern here -- but please clarify, I am happy to answer any questions on this.  I had filed https://github.com/Azure/autorest.csharp/issues/4480 to track updates to ClientUriBuilder -- we have some problems with the current implementation around escaping that will need to be resolved, but if the OpenAI generated client is functional, we can de-prioritize those for the time being.

Reply:
Let me rephrase myself: I compared the new code and old code of this file, because here github is not properly comparing it.
Your change consists of two parts:
1. The first part included in this change: [Azure/autorest.csharp@`e5e175f` (#4519)](https://github.com/Azure/autorest.csharp/pull/4519/commits/e5e175f3717bdb6fce4c2e55e33b022f7699ca03)
2. The second part is that we are removing those variant of `AppendPath` and `AppendQuery` (for instance the overload that takes `DateTimeOffset`). I am wondering that would this causes compile errors in the generated clients? Are you removing those because your specific case is not using those methods, or removing them is part of the issue?